src/rabbit_fifo.erl

```diff
@@ -574,8 +574,9 @@ state_enter(eol, #?MODULE{enqueuers = Enqs,
     [{send_msg, P, eol, ra_event}
      || P <- maps:keys(maps:merge(Enqs, AllConsumers))] ++
         [{mod_call, rabbit_quorum_queue, file_handle_release_reservation, []}];
-state_enter(_, #?MODULE{cfg = #cfg{resource = _Resource} }) ->
-    [{mod_call, rabbit_quorum_queue, file_handle_other_reservation, []}];
+state_enter(State, #?MODULE{cfg = #cfg{resource = _Resource}}) when State =/= leader ->
```

Comment:
This should be the catch all clause, as `leader` and `eol` are already handled. Why do we need the next one? Am I missing something?

Reply:
only because not  having  a catch-all  clause seemed suspicious.

Reply:
It turned out that the clause was used by `state_enter(recover, State)`. Without it all `ct-dead_lettering` QQ tests fail, for example. @kjnilsson @dcorbacho I've left it in, not sure if explicit handling would be worth doing.

Reply:
Right so at the recover stage the State is `undefined` because, well, it hasn't recovered the state yet! Incidentally it will need a file handle reservation for recover so perhaps we should handle this case explicitly. I have to admit I didn't realise the state was `undefined` at recover. This should be clarified in the callback type.

Reply:
oh but I realise we can't reserve at recover as we do not have a queue name available.

Reply:
Ah looking closer it _should_ have the state available at that point as it does read the snapshot or generates the initial state earlier. Ok will need to dig further

Reply:
I just ran ct-dead_lettering without the catch-all and it passed ok so perhaps something else is going on.

Reply:
Same here, not failing either

Reply:
Apart from that this PR looks fine to me